src/pages/ProposalDetailPage.tsx

```diff
@@ -0,0 +1,102 @@
+import { usePdcApi } from '../use-pdc-api';
+import { ProposalDetailPanel } from '../components/ProposalDetailPanel';
+
+interface CanonicalField {
+  id: number;
+  label: string;
+  shortCode: string;
+}
+
+interface Proposal {
+  versions: {
+    version: number;
+    fieldValues: {
+      applicationFormField: {
+        canonicalFieldId: number;
+        position: number;
+      };
+      value: string;
+    }[];
+  }[];
+}
+
+const getValueOfCanonicalField = (
+  canonicalFields: CanonicalField[],
+  proposal: Proposal,
+  canonicalFieldShortCode: string,
+) => {
+  const field = canonicalFields.find(({ shortCode }) => (
+    shortCode === canonicalFieldShortCode
+  ));
+  if (field === undefined) {
+    return undefined;
+  }
+  const fieldValue = proposal.versions[0].fieldValues.find(({ applicationFormField }) => (
+    applicationFormField.canonicalFieldId === field.id
+  ));
+  return fieldValue?.value ?? undefined;
+};
+
+const mapCanonicalFields = (
+  canonicalFields: CanonicalField[],
+  proposal: Proposal,
+) => (
+  proposal.versions[0].fieldValues.map(({ applicationFormField, value }) => ({
+    fieldName: canonicalFields.find(({ id }) => (
+      id === applicationFormField.canonicalFieldId
+    ))?.label ?? 'missing',
+    position: applicationFormField.position,
+    value,
+  }))
+);
+
+const ProposalDetailPage = () => {
+  const params = useParams();
+  const proposalId = params.proposalId ?? 'missing';
+  const canonicalFields = usePdcApi<CanonicalField[]>('/canonicalFields');
+  const proposal = usePdcApi<Proposal>(`/proposals/${proposalId}?includeFieldsAndValues=true`);
+
+  if (canonicalFields === null || proposal === null) {
+    return (
+      <OidcSecure>
+        <ProposalDetailPanel
+          title="Loading..."
+          author="Loading..."
+          authorId="00-0000000"
+          version={0}
+          values={[]}
+        />
+      </OidcSecure>
+    );
+  }
+
+  const title = getValueOfCanonicalField(canonicalFields, proposal, 'proposal_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'proposal_title')
+    ?? `Proposal ${proposalId}`;
+  const author = getValueOfCanonicalField(canonicalFields, proposal, 'organization_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'organization_dba_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'organization_legal_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'proposal_primary_contact_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'proposal_submitter_name')
+    ?? 'Unknown Author';
+  const authorId = getValueOfCanonicalField(canonicalFields, proposal, 'organization_tax_id');
+  const { version } = proposal.versions[0];
+  const values = mapCanonicalFields(canonicalFields, proposal);
+
+  return (
+    <OidcSecure>
+      <ProposalDetailPanel
+        title={title}
+        author={author}
+        authorId={authorId}
+        version={version}
+        values={values}
```

Comment:
I'm a little nervous at how quickly we're going to want to replace this architecture when we really support versions, but you noted that limitation in the description and I think it's the right choice. Interested to know if you already have the adaptations in mind!

Reply:
I do! I'm sure what I have in mind will change a bit as I implement it, but I think I have a good enough plan. And yeah these'll change pretty quick, but I think that's okay!